src/backend/distributed/executor/executor.c

```diff
@@ -2,6 +2,86 @@
  *
  * executor.c
  *
+ * The adaptive executor executes a list of tasks (queries on shards) over
+ * a connection pool per worker node. The results of the queries, if any,
+ * are written to a tuple store.
+ *
+ * Each connection pool (WorkerPool) and each connection (WorkerSession)
```

Comment:
What do you think about adding couple of sentences before this paragraph to give high level definition of WorkerPool and WorkerSession and their relation briefly? Otherwise "connection pool == WorkerPool" is a bit confusing and people will need to look at how these structs are defined and used to make sense of it.

Reply:
Also there is no reference to the pool size part of execution. Is that just a detail or is it one defining characteristic of this executor?

Reply:
Pushed some fixes.